--- feeds/luci/modules/luci-base/luasrc/model/network.lua
+++ feeds/luci/modules/luci-base/luasrc/model/network.lua
@@ -26,1 +26,1 @@
-IFACE_PATTERNS_WIRELESS = { "^wlan%d", "^wl%d", "^ath%d", "^rausb%d", "^rai%d", "^rax%d", "^ra%d", "^wdsi%d", "^wdsx%d", "^wds%d", "^apclii%d", "^apclix%d", "^apcli%d", "^apcliusb%d", "^%w+%.network%d" }
+IFACE_PATTERNS_WIRELESS = { "^wlan%d", "^wl%d", "^ath%d", "^%w+%.network%d" }
--- package/mtk/luci-app-mtwifi/luasrc/controller/mtkwifi.lua
+++ package/mtk/luci-app-mtwifi/luasrc/controller/mtkwifi.lua
@@ -383,9 +383,10 @@
 
 	if iface == "ra0" or iface == "rai0" then
 		if cfgs.EAPifname == "br-lan" then
-			mt_lan = mtkwifi.read_pipe("uci show network.cfg030f15.ports | grep -q "..iface.." && echo 1")
+			mt_lan = mtkwifi.read_pipe("uci get network.lan.ifname | grep -q "..iface.." && echo 1")
 			if mt_lan ~= "1\n" then
-				os.execute("uci add_list network.cfg030f15.ports='"..iface.."'; uci commit network; logger -t MTK-Wi-Fi Add "..iface.." to br-lan; service network reload")
+				new_brlan = string.gsub(mtkwifi.read_pipe("uci get network.lan.ifname"), "\n$", "").." "..iface
+				os.execute("uci set network.lan.ifname='"..new_brlan.."'; uci commit network; logger -t MTK-Wi-Fi Add "..iface.." to br-lan; /etc/init.d/network reload")
 			end
 		end
 	end
@@ -719,16 +720,19 @@
 		__security_cfg(cfgs, vif_idx)
 
 	if vifname == "ra0" or vifname == "rai0" then
-		mt_lan = mtkwifi.read_pipe("uci show network.cfg030f15.ports | grep -q "..vifname.." && echo 1")
+	  mt_lan = mtkwifi.read_pipe("uci get network.lan.ifname | grep -q "..vifname.." && echo 1")
 		if http.formvalue("__lan") == "1" then
 			cfgs.EAPifname = "br-lan"
 			if mt_lan ~= "1\n" then
-				os.execute("uci add_list network.cfg030f15.ports='"..vifname.."'; uci commit network; logger -t MTK-Wi-Fi Add "..vifname.." to br-lan; service network reload")
+				new_brlan = string.gsub(mtkwifi.read_pipe("uci get network.lan.ifname"), "\n$", "").." "..vifname
+				os.execute("uci set network.lan.ifname='"..new_brlan.."'; uci commit network; logger -t MTK-Wi-Fi Add "..vifname.." to br-lan; /etc/init.d/network reload")
 			end
 		else
 			cfgs.EAPifname = ""
 			if mt_lan == "1\n" then
-				os.execute("uci del_list network.cfg030f15.ports='"..vifname.."'; uci commit network; logger -t MTK-Wi-Fi Remove "..vifname.." from br-lan; service network reload")
+				new_brlan = string.gsub(mtkwifi.read_pipe("uci get network.lan.ifname"), "\n$", "")
+				new_brlan = string.gsub(new_brlan, "%s?" .. vifname, "")
+				os.execute("uci set network.lan.ifname='"..new_brlan.."'; uci commit network; logger -t MTK-Wi-Fi Remove "..vifname.." from br-lan; /etc/init.d/network reload")
 			end
 		end
 	end
